AltairZ80/altairz80_sio.c

```diff
@@ -61,14 +61,18 @@ uint8 *URLContents(const char *URL, uint32 *length);
 #define RESULT_LEAD_IN          "URL is not supported on this platform. START URL \""
 #define RESULT_LEAD_OUT         "\" URL END."
 uint8 *URLContents(const char *URL, uint32 *length) {
-    char str[RESULT_BUFFER_LENGTH] = RESULT_LEAD_IN;
+    char str[RESULT_BUFFER_LENGTH];
     char *result;
+    strncpy(str, RESULT_LEAD_IN, strlen(RESULT_LEAD_IN));
+    str[strlen(RESULT_LEAD_IN)] = '\0';

```

Comment:
Why would copying in the string be better than initializing it? And even when this would be better, why not use strcpy, instead of the complicated dance with strncpy and strlen? In any case, with initialization, the compiler can warn when the string doesn't fit; gcc would warn `warning: initializer-string for array of chars is too long`.

Reply:
Strnlen is meant to protect the destination from overrun, but for that to works requires using the size of the destination, not the size of the source, as the length limit.  As written it's merely a complicated way to say "strcpy" with all the hazards of that function.

Reply:
Thanks, @Rhialto, @pkoning2 for pointing these out.  The VS2022 static analysis tool flags the original code with as "str" not being initialized.  I will upload a new patch set using snprintf() that makes the code more readable and should resolve the issue.

Reply:
Hm, if a tool looks at `char str[RESULT_BUFFER_LENGTH] = RESULT_LEAD_IN;` and then complains that `str` is not initialized, I'd have to take the results from this tool with a large amount of salt...

Your other changes seem sensible though, at first glance.